### DISCUSSION TITLE
Export / import the database enumerations?

### DISCUSSION THREAD

**Discussion by meteos77** [textbrowser/biblioteq#151] [Q&A] [answered]
with catmandu, I have automated the CSV import and I use the Enumerations to bring details on some fields

I did not find anything in the 2 configuration files to save the "Database Enumerations".

will there be a way to save and re-import the Database Enumerations ?

my base with my enumerations
[2021-12-27_Base-vierge-avec-données-minimum-de-base.sqlite.zip](https://github.com/textbrowser/biblioteq/files/7780305/2021-12-27_Base-vierge-avec-donnees-minimum-de-base.sqlite.zip)

**Comment by textbrowser** [comment]
There is a reminder for that. See the TO-DO.

**Reply by textbrowser** [reply]
Book Binding Types, Languages, Locations, Monetary Units?

**Comment by meteos77** [comment]
I did try the SQL  ... I didn't succeed all at once but this one for example
SELECT * FROM locations ; it is enough to export to a CSV
[location.csv](https://github.com/textbrowser/biblioteq/files/7780880/location.csv)

but I miss the CSV import template

**Reply by textbrowser** [reply]
I browsed your attached database and those are the categories which have entries. I'm now modifying the import to insert the values.

**Reply by textbrowser** [reply]
![Screenshot_20211227_110438](https://user-images.githubusercontent.com/10701156/147488467-d500ad9d-b484-4176-b8f7-94f9ff4e9839.png)

**Reply by textbrowser** [reply]
The "Book Binding" is not necessary, so you can probably remove it. :)

**Reply by meteos77** [reply]
you are going to remove the "book binding type" field from the BiblioteQ database ?

my script takes the 010$b and transforms the br. into paperback (enumeration)
010 $a 978-2-84666-840-8 $b br. $d 44 EUR $b the 2 vols.

**Reply by textbrowser** [reply]
No. You have an entry "Book Binding" in your database. You can remove it because it doesn't have a functional meaning.

**Reply by meteos77** [reply]
it does create the new enumerations by import.

my initial request was rather to import them into an existing database without creating new documents

I do the 
SQL select to select them
I save the selection in csv

I'm missing the "model-3 import enumeration-type ..." with the right fields (for the moment in the 2 models book and members; the fields don't appear)

**Reply by textbrowser** [reply]
The enumerations are part of the database. They cannot be separate.

**Reply by meteos77** [reply]
I just came back to the version of 15/12 to do a test 

thanks again for this very useful feature
of automatic import of enumerate database !

**Comment by textbrowser** [comment, marked as answer]
Done!

**Reply by meteos77** [reply]
thank you for this feature in summary : 

- the export is done by SQL on each enumeration type
for example for the location : SELECT type,location FROM locations ORDER BY location

- for the import, create a csv file with pseudo books that contain the different enumerations and then delete the books.
 the enumerations will remain in the database

[4-BiblioteQ_fichier-import-énumérations-v2.csv.tar.gz](https://github.com/textbrowser/biblioteq/files/9228170/4-BiblioteQ_fichier-import-enumerations-v2.csv.tar.gz)

 if the .csv file contains new enumerations, they are created automatically.
